lib-rpc/ocamlformat_rpc_lib.ml

```diff
@@ -37,7 +37,16 @@ module type Client_S = sig
   val config :
     (string * string) list -> t -> (unit, [> `Msg of string]) result
 
-  val format : string -> t -> (string, [> `Msg of string]) result
+  val format :
+       ?path:string
+    -> ?config:(string * string) list
+    -> ?overriding_path:string
+    -> ?overriding_config:(string * string) list
```

Comment:
Why not take `format_args` here ?

Reply:
Or maybe only the overriding_path and overriding_config here, I think it's useful for the "formatting the tooltip" workflow.

For the rest I agree that it would be easier to maintain more atomic commands, as long as sending 2 or 3 commands instead of 1 isn't too much overhead:
- sending a Path command when a file from a new dir is formatted (up to vscode to cache the path of the last dir requested)
- sending a Config command if needed (I guess this is mostly needed for the margin)
- sending a Format command

Reply:
> sending a Path command when a file from a new dir is formatted (up to vscode to cache the path of the last dir requested)

How does the client know what's the argument to the Path command? How should vscode decide when to expire the cache of the last dir requested?

> sending a Config command if needed (I guess this is mostly needed for the margin)

In general, how should clients know whether a config is needed? My impression is every formatting needs to use the most recent configuration.

Reply:
> sending 2 or 3 commands instead of 1 isn't too much overhead

Sending many commands is fine, the problem I point out is that there's two workflows: passing arguments to the `Format` command or the `Path` and `Config` commands.
One of the two workflow isn't necessary. Both ways can implement the lib interface but I think the `Format` command with argument is more consistent wrt to the API of `ocamlformat_rpc_lib`.